### ISSUE TITLE
Typeguard not working with non-simple boolean expression

### ISSUE THREAD

**Issue by maximelkin** [Microsoft/TypeScript#22264]
typescript Version:**  2.7.2

**Search Terms:** typeguard type guard if typeof false

**Code**

```ts
function test1(a: number | string) {
    if ((typeof  a === 'number') === false) {
        return
    }
    return a.toFixed() // here error
}
function test2(a: number | string) {
    if (!(typeof  a === 'number')) {
        return
    }
    return a.toFixed()
}
```

**Expected behavior:** no error in test1

**Actual behavior:** error in test1

**Playground Link:** https://www.typescriptlang.org/play/index.html#src=function%20test1(a%3A%20number%20%7C%20string)%20%7B%0D%0A%20%20%20%20if%20((typeof%20%20a%20%3D%3D%3D%20'number')%20%3D%3D%3D%20false)%20%7B%0D%0A%20%20%20%20%20%20%20%20return%0D%0A%20%20%20%20%7D%0D%0A%20%20%20%20return%20a.toFixed()%20%2F%2F%20here%20error%0D%0A%7D%0D%0Afunction%20test2(a%3A%20number%20%7C%20string)%20%7B%0D%0A%20%20%20%20if%20(!(typeof%20%20a%20%3D%3D%3D%20'number'))%20%7B%0D%0A%20%20%20%20%20%20%20%20return%0D%0A%20%20%20%20%7D%0D%0A%20%20%20%20return%20a.toFixed()%0D%0A%7D

**Comment by andy-ms** [comment]
I've seen an issue exactly like this before but can't seem to find it. We don't support writing `=== false` or `=== true` as a type guard.

**Comment by maximelkin** [comment]
Thanks for reply
